src/util/persistent_logger.js

```diff
@@ -208,6 +209,146 @@ class PersistentLogger {
         }
     }
 
+    /**
+     * approx_entries returns approximate number of enties in the current active file
+     * based on the chosen strategy and sample size
+     * @param {{
+     *  strategy?: "TOP_K" | "MID_K" | "MIXED_K",
+     *  samples?: number
+     * }} cfg
+     * @returns {Promise<number>}
+     */
+    async approx_entries(cfg) {
```

Comment:
Split into 2 - approx entries (without these strategies - configure to load how many pages to load and then maybe a randomized approximation).

Reply:
Consider evenly distributed offsets + randomized offsets.